api-registration/server/index.js

```diff
@@ -0,0 +1,60 @@
+require('dotenv/config');
+const pg = require('pg');
+const argon2 = require('argon2'); // eslint-disable-line
+const express = require('express');
+const ClientError = require('./client-error');
+const errorMiddleware = require('./error-middleware');
+
+const db = new pg.Pool({ // eslint-disable-line
+  connectionString: process.env.DATABASE_URL,
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+const app = express();
+
+const jsonMiddleware = express.json();
+
+app.use(jsonMiddleware);
+
+app.post('/api/auth/sign-up', (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) {
+    throw new ClientError(400, 'username and password are required fields');
+  }
+  argon2.hash(username)
+    .then(hashedPassword => {
+      const sql = `
+    insert into "users" ("username", "hashedPassword")
+    values ($1, $2)
+    returning *
```

Comment:
Here you can specify the columns you want returned, then you don't have to do the extra work before sending the data to the client.
```JAVASCRIPT
const sql = `
    insert into "users" ("username", "hashedPassword")
    values ($1, $2)
    returning "userId", "createdAt", "username"
`;